Simplenote/SPTracker+Extensions.swift

```diff
@@ -27,6 +27,58 @@ extension SPTracker {
     static func trackVerificationDismissed() {
         trackAutomatticEvent(withName: "verification_dismissed", properties: nil)
     }
+
+    static func trackNotice(_ value: String) {
+        trackAutomatticEvent(withName: "notice_presented", properties: ["notice": value])
+    }
+
+    static func trackNoticeAction(_ value: String, action: String) {
+        trackAutomatticEvent(withName: "notice_action_tapped", properties: ["notice": value, "notice_action": action])
+    }
+
+    static func trackNoticeLinkCopied() {
+        trackNotice("Link Copied")
+    }
+
+    static func trackNoticePublishing() {
+        trackNotice("Publishing")
+    }
+
+    static func trackNoticeUnpublishing() {
+        trackNotice("Unpublishing")
+    }
+
+    static func trackNoticeNoteTrashed() {
+        trackNotice("Note Trashed")
+    }
+
+    static func trackNoticeNotesTrashed() {
+        trackNotice("Notes Trashed")
+    }
+
+    static func trackNoticeUnpublished() {
+        trackNotice("Unpublished")
+    }
+
+    static func trackNoticePublished() {
+        trackNotice("Published")
+    }
+
+    static func trackNoticeActionTrashUndo() {
+        trackNoticeAction("Note Trashed", action: "Undo")
+    }
+
+    static func trackNoticeActionMultipleTrashUndo() {
+        trackNoticeAction("Notes Trashed", action: "Undo")
+    }
+
+    static func trackNoticeActionUnpublished() {
+        trackNoticeAction("Unpublished", action: "Undo")
+    }
+
+    static func trackNoticeActionPublished() {
+        trackNoticeAction("Published", action: "Copy Link")
+    }
```

Comment:
Since all of the code that invoke these API(s) is pure Swift, how about something like this?

```
enum NoticeKind: String {
   case trashed
   case published
   case unpublished
   //..
}

enum NoticeActionKind {
     case undo
     case interlinkCopied
     case publicLinkCopied
}

func trackPresentedNotice(kind: NoticeKind) {
   // ..
}

func trackPerformedAction(noticeAction: NoticeActionKind, noticeKind: NoticeKind ) {
  // ..
}
```


Naming can probably be improved, but you'd be able to get rid of similar looking methods. Maybe?

Reply:
I like it!